src/mds/Server.cc

```diff
@@ -130,6 +130,9 @@ void Server::dispatch(Message *m)
 	  dout(3) << "queuing completed op" << dendl;
 	  queue_replay = true;
 	}
+	// this request was created before the cap reconnect message, drop any embedded
+	// cap releases.
+	req->releases.clear();
```

Comment:
This inline is for kernel client. the kernel client sends requests before composing the cap reconnect message, but it does not clear the embedded cap releases. 

Even without this line, MDS should work smoothly. because Locker::process_request_cap_release() and Capability::confirm_receipt() can detect out-of-date cap releases.